Goal: Prepare the sources to better respect FHS

Fixes: #49011

Status wrt upstream: Everything is applied as of 3.4.0

Note: The last remaining bit is the location of "private files
      We historically have them in /var/lib/samba while upstream
      has them in /etc/samba
      We need to provide a migraiton path and go back to the "normal"
      file layout

Index: samba-3.3.0rc2/source/passdb/pdb_tdb.c
===================================================================
--- samba-3.3.0rc2.orig/source/passdb/pdb_tdb.c
+++ samba-3.3.0rc2/source/passdb/pdb_tdb.c
@@ -1613,7 +1613,7 @@
 	/* save the path for later */
 
 	if (!location) {
-		if (asprintf(&tdbfile, "%s/%s", lp_private_dir(),
+		if (asprintf(&tdbfile, "%s/%s", get_dyn_STATEDIR(),
 			     PASSDB_FILE_NAME) < 0) {
 			return NT_STATUS_NO_MEMORY;
 		}
Index: samba-3.3.0rc2/source/passdb/secrets.c
===================================================================
--- samba-3.3.0rc2.orig/source/passdb/secrets.c
+++ samba-3.3.0rc2/source/passdb/secrets.c
@@ -59,7 +59,7 @@
 		return True;
 
 	fname = talloc_asprintf(talloc_tos(), "%s/secrets.tdb",
-				lp_private_dir());
+				get_dyn_STATEDIR());
 	if (fname == NULL) {
 		return false;
 	}
@@ -1113,7 +1113,7 @@
 	TDB_DATA vers;
 	uint32 ver;
 	TDB_CONTEXT *tdb_sc = NULL;
-	char *fname = talloc_asprintf(mem_ctx, "%s/schannel_store.tdb", lp_private_dir());
+	char *fname = talloc_asprintf(mem_ctx, "%s/schannel_store.tdb", get_dyn_STATEDIR());
 
 	if (!fname) {
 		return NULL;
